src/components/asset-list/RecyclerAssetList2/LegacyWrappedTokenFamilyHeader.tsx

```diff
@@ -0,0 +1,211 @@
+import { useLatestCallback } from '@/hooks';
+import { useRemoteConfig } from '@/model/remoteConfig';
+import { NFTS_ENABLED, useExperimentalFlag } from '@/config';
+import lang from 'i18n-js';
+import React, { useMemo, useRef } from 'react';
+import { Animated, Easing, StyleSheet, View } from 'react-native';
+import FastImage from 'react-native-fast-image';
+import CaretImageSource from '@/assets/family-dropdown-arrow.png';
+import { ButtonPressAnimation } from '@/components/animations';
+import TokenFamilyHeaderIcon from '@/components/token-family/TokenFamilyHeaderIcon';
+import { Text } from '@/design-system';
+import { ImgixImage } from '@/components/images';
+import { ThemeContextProps } from '@/theme';
+import { useOpenCollectionsStore } from '@/state/nfts/openCollectionsStore';
+
+type Props = {
+  name: string;
+  total?: number;
+  image?: string;
+  theme: ThemeContextProps;
+  testID?: string;
+};
+
+export default React.memo(function WrappedTokenFamilyHeader({ name, total, image, theme, testID }: Props) {
+  const { nfts_enabled } = useRemoteConfig();
+  const nftsEnabled = useExperimentalFlag(NFTS_ENABLED) || nfts_enabled;
+
+  const nameKey = `${name}-legacy`;
+
+  const isFamilyOpen = useOpenCollectionsStore(state => state.isCollectionOpen(nameKey));
+
+  const handleToggle = useLatestCallback(() => useOpenCollectionsStore.getState().toggleCollection(nameKey));
+
+  if (!nftsEnabled) return null;
+
+  return (
+    <TokenFamilyHeader
+      childrenAmount={total}
+      familyImage={image}
+      isOpen={isFamilyOpen}
+      onPress={handleToggle}
+      testID={testID}
+      theme={theme}
+      title={name}
+    />
+  );
+});
+
+export const TokenFamilyHeaderAnimationDuration = 200;
+export const TokenFamilyHeaderHeight = 50;
+
+const sx = StyleSheet.create({
+  amountContainer: {
+    marginRight: 10,
+  },
+  center: {
+    alignItems: 'center',
+    flexDirection: 'row',
+    justifyContent: 'center',
+  },
+  chevron: {
+    height: 18,
+    width: 8,
+  },
+  content: {
+    alignItems: 'center',
+    flexDirection: 'row',
+    height: TokenFamilyHeaderHeight,
+    justifyContent: 'space-between',
+    padding: 19,
+    paddingRight: 14,
+    width: '100%',
+  },
+  title: {
+    flex: 1,
+    paddingRight: 9,
+  },
+});
+
+type TokenFamilyHeaderProps = {
+  childrenAmount?: number;
+  emoji?: string;
+  familyImage?: string;
+  isOpen?: boolean;
+  onPress?: () => void;
+  testID?: string;
+  title: string;
+  theme: ThemeContextProps;
+};
+
+const TokenFamilyHeader = ({ childrenAmount, emoji, familyImage, isOpen, onPress, testID, title, theme }: TokenFamilyHeaderProps) => {
+  const { colors } = theme;
+
+  const toValue = Number(!!isOpen);
+
+  const aRef = useRef<{
+    animation: Animated.Value;
+    toValue: number;
+    isRunning: boolean;
+  } | null>(null);
+
+  // HACK ALERT
```

Comment:
mutating refs during render will generally cause confusing issues, i think just converting this to state would be great - you can do a setState() inside the render as well though i wonder if reading toValue during render is a bad practice as it's also not react state, ideally you'd have another state here with currentValue and read that